### PR TITLE
fix: Update `Object.prototype.__proto__` related comments

### PR DESCRIPTION
**Pull request by hyp3rflow**
As https://github.com/denoland/deno/pull/14864#discussion_r897439077 mentioned, Update comments pointing `Object.prototype.__proto__` in ECMA262
Related ECMA262 PR: https://github.com/tc39/ecma262/pull/2125

### CODE REVIEW

File: cli/tsc/99_main_compiler.js [hunk 1]
```diff
@@ -9,7 +9,7 @@
 // instances convert TypeScript to JavaScript.
 
 // Removes the `__proto__` for security reasons.  This intentionally makes
-// Deno non compliant with ECMA-262 Annex B.2.2.1
+// Deno non compliant with ECMA-262 20.1.3.8
```
Comment (littledivy) [inline cli/tsc/99_main_compiler.js:12]: How is Deno non-compliant if the `__proto__` accessor is normative optional?

Comment (hyp3rflow) [reply cli/tsc/99_main_compiler.js:12]: oops! How about just removing it? @littledivy 
```suggestion
```

Comment (littledivy) [reply cli/tsc/99_main_compiler.js:12]: that works!

Comment (hyp3rflow) [reply cli/tsc/99_main_compiler.js]: I've fixed this and decided not to change comment in testdata! thanks :) @littledivy

**Review by littledivy — APPROVED** [review]

**Pull request merged**